### ISSUE TITLE
Console warning when using variables should not include quotation marks in solution

### ISSUE THREAD

**Issue by knoll3** [fish-shell/fish-shell#8198]
When trying to use a bash variable in fish, fish returns this error and suggestion:

```
$ some-command $(SOME_VARIABLE)
fish: $(...) is not supported. In fish, please use '(SOME_VARIABLE)'.
some-command $(SOME_VARIABLE)
```
I think this suggestion would be more clear if it didn't include quotation marks for the variable.
```
$ some-command $(SOME_VARIABLE)
fish: $(...) is not supported. In fish, please use (SOME_VARIABLE).
some-command $(SOME_VARIABLE)
```

$ fish --version
fish, version 3.2.2

$ uname -a
Linux kyle-Latitude-5591 5.8.0-59-generic #66~20.04.1-Ubuntu SMP Thu Jun 17 11:14:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by zanchey** [comment]
I see there's an inconsistency here where the first bit of syntax doesn't have quotation marks but the second does, so I think making them consistent makes sense.

**Comment by mqudsi** [comment]
Good point. I agree especially since a newcomer to fish wouldn't be expected to know the quoting behavior and what is expanded vs what is kept literally in a single- or double-quoted context.

**Comment by mqudsi** [comment]
I wonder if there's some ANSI voodoo equivalent of `<code>foo</foo>` that we could use. A literal backtick like in markdown would be confused with bash's stupid shortcut for command substitution and any of the quotes would risk changing the meaning if taken literally. An obvious option is just changing the color? Italics in the terminal are ugly (and poorly supported), but bold is an option?

Note that removing the quotes for this specific case introduces an inconsistency with the other warnings when bash-isms are detected, but those are less egregious since they typically involve some words before or after the part of the expression that has changed... which is also an option here, e.g. 

```
$ printf $(foo)
fish: Substitutions like "echo $(foo)" are not supported, please use "echo (foo)" instead
```

(i.e. using an example invocation rather than just the expression in question, but I'm not sure how much I like that.)

It's probably best to just drop the quotes for now.

**Comment by faho** [comment]
@mqudsi Note that this specific error is no longer an error - #8059.

The errors this affects are in parse_constants.h. We quote the code parts for some messages, and I'd argue for something like `'continue' while not inside of loop` it's necessary because "continue" is an ordinary english word. When it's sigils and punctuation like `$fish_pid` it's not as necessary, and we skip it in those cases.

At least that's the justification for the current situation I can come up with?

**Comment by mqudsi** [comment]
Yeah, I was hunting for it then I realized it was obviously dropped by the recent new features. We really need to clean up the `po/` files - is there an automated tool for that?

**Comment by faho** [comment]
See https://github.com/fish-shell/fish-shell/blob/master/CONTRIBUTING.rst#message-translations:

```fish
msgmerge --update --no-fuzzy-matching po/LANG.po messages.pot
```

Important: Fuzzy matching is turned off because we've seen that break format strings before, leading to crashes.

Honestly I've been neglecting the localizations because we don't really have enough people contributing for them to be useful, so putting in work there makes relatively little sense.
